discovery-provider/src/tasks/tracks.py

```diff
@@ -172,6 +173,102 @@ def update_remixes_table(session, track_record, track_metadata):
+    if track_record.title == track_metadata['title']:
+        return
+
+    # Get the title slug, and set the new slug to that
+    # (will check for conflicts later)
+    new_track_slug_title = (
+        helpers.create_track_slug(track_metadata['title'])
+    )
+    new_track_slug = new_track_slug_title
+
+    # Find the current route for the track
+    prev_track_route_record = (
+        session
+        .query(TrackRoute)
+        .filter(TrackRoute.track_id == track_record.track_id,
+                TrackRoute.is_current == True)  # noqa: E712
+        .one_or_none()
+    )
+
+    if prev_track_route_record is not None:
+        if prev_track_route_record.title_slug == new_track_slug_title:
+            # If the title slug hasn't changed, we have no work to do
+            return
+        # The new route will be current
+        prev_track_route_record.is_current = False
+
+    # Check for conflicts Find the max conflict ID for the track slug
+    max_collision_id = (
+        session
+        .query(functions.max(TrackRoute.collision_id))
+        .filter(TrackRoute.title_slug == new_track_slug_title,
+                TrackRoute.owner_id == track_record.owner_id)
+        .one_or_none()
+    )[0]
+
+    new_collision_id = 0
+    has_collisions = False
+    if max_collision_id is not None:
+        has_collisions = True
+        new_collision_id = max_collision_id
+    while has_collisions:
+        # If there is an existing track by the user with that slug,
+        # then we need to append the collision number to the slug
+        new_collision_id += 1
+        new_track_slug = helpers.create_track_slug(
+            track_metadata["title"],
+            new_collision_id)
+
+        # Check for new collisions after making the new slug
+        # In rare cases the user may have track names that end in numbers that
+        # conflict with this track name when the collision id is appended,
+        # for example they could be trying to create a route that conflicts
+        # with the old routing (of appending -{track_id}) This is a fail safe
+        # to increment the collision ID until no such collisions are present.
+        #
+        # Example scenario:
+        #   - User uploads track titled "Track" (title_slug: 'track')
+        #   - User uploads track titled "Track 1" (title_slug: 'track-1')
+        #   - User uploads track titled "Track" (title_slug: 'track')
+        #       - Try collision_id: 1, slug: 'track-1' and find new collision
+        #       - Use collision_id: 2, slug: 'track-2'
+        #   - User uploads track titled "Track" (title_slug: 'track')
+        #       - Use collision_id: 3, slug: 'track-3'
+        #   - User uploads track titled "Track 1" (title_slug: 'track-1')
+        #       - Use collision_id: 1, slug: 'track-1-1'
+        #
+        # This may be expensive with many collisions, but should be rare.
+        collision_count = (
```

Comment:
I know you've thought this through way more than I have, but for my own sanity - would there be any benefit here to picking the max of the new slug's collision IDs, rather than just incrementing by 1 at the top of the loop?

Reply:
That's a good thought. I did have to think about it for a minute but I think it won't help since the new slug collision won't have higher collision IDs.

Like in the example in the comment:
        #   - User uploads track titled "Track" (title_slug: 'track')
        #   - User uploads track titled "Track 1" (title_slug: 'track-1')
        #   - User uploads track titled "Track" (title_slug: 'track')
        #       - Try collision_id: 1, slug: 'track-1' and find new collision **with existing `track-1` (title_slug: `track-1` collision_id: 0)**
        
Any other tracks with the title_slug `track-1` would have an appended collision ID (eg `track-1-1`), so they're not in conflict with `track-2`. If there exists a `track-2` it will have a title_slug of `track-2` and collision ID of 0, and if there were multiple they'd be `track-2-1` etc so that wouldn't conflict. And it would continue for `track-3` etc. So there's not really a way to shortcut it. If there were more `track-2`s and `track-3`s that were title_slug `track` then they would have been covered in the first `max_collision_id`